server/api/organisation/organisation.controller.js

```diff
@@ -46,7 +46,7 @@ const putOrganisation = async (req, res) => {
   await Organisation.findByIdAndUpdate(req.params._id, { $set: req.body })
   res.json(await Organisation.findById(req.params._id).exec())
   // Domain string validation
-  if (Object.keys(Organisation).includes('domainName') && !orgValidation.domainName(req.body.domainName)) {
+  if (Object.keys(res.body).includes('domainName') && !orgValidation.domainName(req.body.domainName)) {
```

Comment:
should be **req.body** not res. body. 

the idea is that you are about to call orgValidation.domainName on the request body domainName.  But you don't want to do this if it is not in the request.  So you might want to check req.body.domainName is not Null.  but if the field is not present this would throw an exception.  So instead we check whether it is in the object keys. Thats the list of properties that the object actually has.